packages/api/cms-api/src/page-tree/blocks/internal-link-block-transformer.service.ts

```diff
@@ -0,0 +1,50 @@
+import { BlockTransformerServiceInterface } from "@comet/blocks-api";
+import { Injectable } from "@nestjs/common";
+import { ModuleRef } from "@nestjs/core";
+
+import { PageTreeReadApiService } from "../page-tree-read-api.service";
+import type { InternalLinkBlockData } from "./internal-link.block";
+
+type TransformResponse = {
+    targetPage: {
+        id: string;
+        name: string;
+        path: string;
+        documentType: string;
+    } | null;
+    targetPageAnchor?: string;
+};
+
+@Injectable()
+export class InternalLinkBlockTransformerService implements BlockTransformerServiceInterface<InternalLinkBlockData, TransformResponse> {
+    constructor(private moduleRef: ModuleRef) {}
+
+    async transformToPlain(block: InternalLinkBlockData) {
+        if (!block.targetPageId) {
+            return {
+                targetPage: null,
+            };
+        }
+
+        // @nsams: Can't get the PageTreeReadApiService using constructor injection
+        const pageTreeReadApiService = await this.moduleRef.resolve(PageTreeReadApiService);
```

Comment:
@nsams I can't inject the PageTreeReadApiService using the constructor, it is always undefined. I assume this is because of injecting `CONTEXT`, making the service request-scoped. Should we support request-scoped services here as well?

Reply:
> Is it doable?

When quickly trying to replace `moduleRef.get` with `moduleRef.resolve`, I didn't get it to work. But I can attempt it once more

Reply:
Well, this was kinda hard 😓 https://github.com/vivid-planet/comet/pull/1946/commits/934c9273a637788634cd2f222ad2daf5b72a1f60